dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/CommonTaskProcessor.java

```diff
@@ -171,4 +177,25 @@ private void killRemoteTask() throws ExecuteException {
 
         nettyExecutorManager.executeDirectly(executionContext);
     }
+
+    protected void checkAndReplaceTestDataSource() {
```

Comment:
Is it possible to replace the datasource when creating a task instance ?

Reply:
In my first version, I replaced it when creating TaskInstance. Since only tasks of common type are sent to the worker, including tasks with bound data sources, such as SQL, and tasks without bound data sources, such as shell, the data source is replaced here, and the control task type has been fine-grained.